src/coinbase/wallet.ts

```diff
@@ -60,26 +60,58 @@ export class Wallet {
    * Instead, use User.createWallet.
    *
    * @constructs Wallet
+   * @param intervalSeconds - The interval at which to poll the CDPService, in seconds.
+   * @param timeoutSeconds - The maximum amount of time to wait for the ServerSigner to create a seed, in seconds.
    * @throws {ArgumentError} If the model or client is not provided.
    * @throws {InternalError} - If address derivation or caching fails.
    * @throws {APIError} - If the request fails.
    * @returns A promise that resolves with the new Wallet object.
    */
-  public static async create(): Promise<Wallet> {
+  public static async create(intervalSeconds = 0.2, timeoutSeconds = 20): Promise<Wallet> {
     const result = await Coinbase.apiClients.wallet!.createWallet({
       wallet: {
         network_id: Coinbase.networkList.BaseSepolia,
+        use_server_signer: Coinbase.useServerSigner,
       },
     });
 
     const wallet = await Wallet.init(result.data, undefined, []);
 
+    if (Coinbase.useServerSigner) {
+      await wallet.waitForSigner(wallet.getId()!, intervalSeconds, timeoutSeconds);
+    }
+
     await wallet.createAddress();
     await wallet.reload();
 
     return wallet;
   }
 
+  /**
+   * Waits until the ServerSigner has created a seed for the Wallet.
+   *
+   * @param walletId - The ID of the Wallet that is awaiting seed creation.
+   * @param intervalSeconds - The interval at which to poll the CDPService, in seconds.
+   * @param timeoutSeconds - The maximum amount of time to wait for the ServerSigner to create a seed, in seconds.
+   * @throws {APIError} if the API request to get a Wallet fails.
+   * @throws {Error} if the ServerSigner times out.
+   */
+  private async waitForSigner(
+    walletId: string,
+    intervalSeconds = 0.2,
+    timeoutSeconds = 20,
+  ): Promise<void> {
+    const startTime = Date.now();
+    while (Date.now() - startTime < timeoutSeconds * 1000) {
+      const response = await Coinbase.apiClients.wallet!.getWallet(walletId);
+      if (response?.data.server_signer_status === ServerSignerStatus.ACTIVE) {
+        return;
+      }
+      await delay(intervalSeconds);
+    }
```

Comment:
can we have a generic util function for this logic? we use the exactly same in address class.

Reply:
Yes lets follow up with this change since it'll add scope outside of server signer support to change in transfers etc

Reply:
+1 to splitting out the logic however